core/src/main/java/com/sequenceiq/cloudbreak/service/eventbus/CloudResourcePersisterService.java

```diff
@@ -65,8 +68,10 @@ public ResourceNotification retrieve(ResourceNotification data) {
         return null;
     }
 
-    private StackRepository getStackRepository() {
-        return getRepositoryForEntity(Stack.class);
+    private Stack getStackRepository(Long stackId) {
+        StackRepository stackRepository = getRepositoryForEntity(Stack.class);
+        Optional<Stack> stackOpt = stackRepository.findById(stackId);
+        return stackOpt.orElseThrow(() -> new NotFoundException(String.format("Stack had not been found with id: '%s'", stackId)));
```

Comment:
I prefer to create a shorten version of those `orElseThrow(() -> new NotFoundException(` like `orElseThrow(notFound("Stack", stackId))` everywhere.